controllers/dockermachine_controller.go

```diff
@@ -115,6 +115,12 @@ func (r *DockerMachineReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 		return ctrl.Result{}, err
 	}
 
+	//delete dockerMachine
+	if !machine.ObjectMeta.DeletionTimestamp.IsZero() {
+		log.Info("Start delete dockerMachine")
+		return r.delete(ctx, cluster, machine, dockerMachine)
+	}
+
 	// creating loadbalancer
```

Comment:
It doesn't look like we are leveraging a finalizer today, in order for delete to function a finalizer will need to be added here if not already present, similar to: https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/pkg/controller/awsmachine/awsmachine_controller.go#L211-L214

Reply:
Thank you for your guidance. 
I will add a finalizer first